query-engine/prisma-models/src/field/mod.rs

```diff
@@ -214,7 +214,7 @@ impl std::fmt::Display for TypeIdentifier {
             TypeIdentifier::Float => write!(f, "Float"),
             TypeIdentifier::Decimal => write!(f, "Decimal"),
             TypeIdentifier::Boolean => write!(f, "Bool"),
-            TypeIdentifier::Enum(e) => write!(f, "Enum{e}"),
+            TypeIdentifier::Enum(e) => write!(f, "Enum{e:?}"),
```

Comment:
Same thing here. `TypeIdentifier::to_string()` is used in two places in the schema builder to build schema type names. Here:
- https://github.com/prisma/prisma-engines/blob/df61b5f5ab3f27359e273f1cba4e5a9c7cb8485a/query-engine/schema-builder/src/input_types/fields/field_filter_types.rs#L209
- https://github.com/prisma/prisma-engines/blob/df61b5f5ab3f27359e273f1cba4e5a9c7cb8485a/query-engine/schema-builder/src/input_types/fields/field_filter_types.rs#L255

This is also breaking. Let me know if you need ideas as to how we make this backward-compatible. Two quick ideas without digging much:

- Don't refactor `TypeIdentifier` yet and keep passing the enum name for now.
- Use the `Zipper` so that you can render the `Enum` name.

In case it's not clear: Changing any schema type name is a breaking change because those types are eventually part of the public API of the Client. Every single schema type is publicly importable. Changing the name would break those imports.

In this case, not only is it breaking, but it would harm the DX since we wouldn't be able to know which type corresponds to which enum anymore. It is critical that we keep the enum names in their respective schema type names.